src/services/CertificateGenerationService.ts

```diff
@@ -1,898 +1,879 @@
+                            : undefined,
+                    ExpiryDate: testResult.testTypes.testExpiryDate,
+                    AtfNameAtfPNumber:
+                        testResult.testStationName + " " + testResult.testStationPNumber,
+                    Notes: testResult.testTypes.additionalNotesRecorded,
+                    TestTypeDate: testResult.testTypes.testTypeStartTimestamp,
+                };
+
+                console.log("CHECK HERE DOCGENPAYLOAD -> ", docGenPayloadAdr);
+
+                return docGenPayloadAdr;
+        }
+    }
 
-        return docGenPayloadAdr;
+    /**
+     * Retrieves the adrDetails from a techRecord searched by vin
+     * @param testResult - testResult from which the VIN is used to search a tech-record
+     */
+    public getAdrDetails = async (testResult: any) => {
+        const searchRes = await this.callSearchTechRecords(testResult.vin);
+        return await this.processGetCurrentProvisionalRecords(searchRes) as TechRecordType<"hgv" | "trl">;
     }
-  }
-
-  /**
-   * Retrieves the adrDetails from a techRecord searched by vin
-   * @param testResult - testResult from which the VIN is used to search a tech-record
-   */
-  public async getAdrDetails(testResult: any) {
-    const techRecord = await this.getTechRecord(testResult);
-
-    return techRecord.techRecord[0].adrDetails;
-  }
-
-  /**
-   * Returns true if an adrDetails object contains a tankDetails object
-   * @param testResult - testResult from which the VIN is used to search a tech-record
-   */
-  public containsTankDetails(adrDetails: any) {
-    return adrDetails && adrDetails.tank && adrDetails.tank.tankDetails;
-  }
-
-  /**
-   * Retrieves the vehicle weight details for Roadworthisness certificates
-   * @param testResult
-   */
-  public async getWeightDetails(testResult: any) {
-    const result = await this.getTechRecord(testResult);
-    if (result) {
-      console.log("techRecord for weight details found");
-      const weightDetails: IWeightDetails = {
-        dgvw: result.techRecord[0].grossDesignWeight,
-        weight2: 0,
-      };
-      if (testResult.vehicleType === VEHICLE_TYPES.HGV) {
-        weightDetails.weight2 = result.techRecord[0].trainDesignWeight;
-      } else {
-        if (
-          result.techRecord[0].axles &&
-          result.techRecord[0].axles.length > 0
-        ) {
-          const initialValue = 0;
-          weightDetails.weight2 = result.techRecord[0].axles.reduce(
-            (
-              accumulator: number,
-              currentValue: { weights: { designWeight: number } }
-            ) => accumulator + currentValue.weights.designWeight,
-            initialValue
-          );
+
+    /**
+     * Used to unflatten a specific object inside of flat technical record
+     * @param obj
+     * @param keyFilter
+     */
+    public unflattenFlatObject<T extends { [key: string]: any }>(obj: T, keyFilter: string): NestedObject<T> {
```

Comment:
I guess my question around the `unflattenFlatObject` is why are we using it? If the idea of the flat model is that it's easier to use, why are we converting back to this old nested model?

Reply:
I'm using it to identify if certain section of the flat object is present and keep the logic the same for example where previously it checks if tankDetails are present or tc2Details are present in the object. With the flat structure, it can't check without listing out every key but at that point because it's in a flat structure, you wouldn't be able to check if tank is present or tank details without listing the entire key (techRecord_adrDetails_tank_tankDetails_tc2Details_tc2IntermediateApprovalNo).